src/torrent/net/resolver.cc

```diff
@@ -44,15 +44,15 @@ Resolver::resolve_preferred(void* requester, const std::string& hostname, int fa
       if (err == 0) {
         if (sin != nullptr && sin6 != nullptr) {
           if (preferred == AF_INET)
-            result = sa_copy_in(sin.get());
+            result = std::reinterpret_pointer_cast<sockaddr>(sin);
```

Comment:
Copying is required here.

Reply:
No, it shares the underlying use count.

However my real point is that a copy is intentionally done here as we're passing data from one thread to another, and the source thread does it's own cleanup.

There's no reason to try to optimize this since we want to be as safe as possible.